debugtools/DDR_VM/src/com/ibm/j9ddr/corereaders/elf/ELFRISCV64DumpReader.java

```diff
@@ -0,0 +1,147 @@
+{
+
+	protected ELFRISCV64DumpReader(ELFFileReader reader) throws IOException, InvalidDumpFormatException
+	{
+		super(reader);
+	}
+
+	/* (non-Javadoc)
+	 * @see com.ibm.j9ddr.corereaders.elf.ELFDumpReader#readUID()
+	 */
+	@Override
+	protected long readUID() throws IOException
+	{
+		return _reader.readInt() & 0xffffffffL;
+	}
+
+	@Override
+	protected String getProcessorType()
+	{
+		return "riscv64";
+	}
+
+	@Override
+	protected long getBasePointerFrom(Map<String, Number> registers)
+	{
+		return getStackPointerFrom(registers);
+	}
+
+	@Override
+	protected long getInstructionPointerFrom(Map<String, Number> registers)
+	{
+		return registers.get("pc").longValue();
+	}
+
+	@Override
+	protected long getLinkRegisterFrom(Map<String, Number> registers)
+	{
+		return registers.get("ra").longValue();
+	}
+
+	@Override
+	protected String getStackPointerRegisterName()
+	{
+		return "sp";
+	}
+
+	@Override
+	protected String[] getDwarfRegisterKeys()
+	{
+		/* Need to consult the platform ABI to map register names to dwarf numbers */
+		String[] registerNames = new String[65];
+		
+		/* Set the default register+n strings */
+		for( int i = 0; i < 3; i++) {
+			registerNames[5 + i] = "t" + i;			/* Temporaries */
+		}
+		
+		for( int i = 0; i < 8; i++) {
+			registerNames[10 + i] = "a" + i;		/* Function arguments */
+			registerNames[32 + i] = "ft" + i;		/* FP temporaries */
+			registerNames[42 + i] = "fa" + i;		/* FP arguments */
+		}
+		
+		for( int i = 0; i < 10; i++) {
+			registerNames[18 + i] = "s" + 1 + i;	/* Saved registers */
+			registerNames[50 + i] = "fs" + 2 + i;	/* FP saved registers */
+		}
+		
+		for( int i = 0; i < 4; i++) {
+			registerNames[28 + i] = "t" + 3 + i;	/* Saved registers */
+			registerNames[60 + i] = "ft" + 8 + i;	/* FP temporaries */
+		}
+		
+		for( int i = 0; i < 2; i++) {
+			registerNames[8 + i] = "s" + i;			/* Saved registers */
+			registerNames[40 + i] = "fs" + i;		/* FP saved registers */
+		}
+		
+		/* Set the registers with special names */
+		registerNames[0] = "pc";		/* Hard-wired zero or Program counter */
+		registerNames[1] = "ra";		/* Return address */
+		registerNames[2] = "sp";		/* Stack pointer */
+		registerNames[3] = "gp";		/* Global pointer */
+		registerNames[4] = "tp";		/* Thread pointer */
+		registerNames[64] = "fcsr";		/* Floating-Point Control and Status Register */
+		
+		return registerNames;
+	}
+	
+	@Override
+	protected SortedMap<String, Number> readRegisters() throws IOException
+	{
+		String[] registerNames = { "pc", "ra", "sp", "gp", "tp", "t0", "t1", "t2", "s0", "s1",
+				"a0", "a1", "a2", "a3", "a4", "a5", "a6", "a7", "s2", "s3",
+				"s4", "s5", "s6", "s7", "s8", "s9", "s10", "s11", "t3", "t4",
+				"t5", "t6", "ft0", "ft1", "ft2", "ft3", "ft4", "ft5", "ft6", "ft7",
+				"fs0", "fs1", "fa0", "fa1", "fa2", "fa3", "fa4", "fa5", "fa6", "fa7",
+				"fs2", "fs3", "fs4", "fs5", "fs6", "fs7", "fs8", "fs9", "fs10", "fs11",
+				"ft8", "ft9", "ft10", "ft11", "fcsr" };
```

Comment:
This looks like the same array as below in `ArchRISCV64` and computed above in `getDwarfRegisterKeys()`: it should appear one place and be reused.

A more compact expression is
```
String regNames = 
  ("pc,ra,sp,gp,tp,t0,t1,t2,s0,s1,a0,a1,a2,a3,a4,a5,a6,a7,s2,s3,s4,s5,s6,s7,s8,s9,s10,s11"
+ ",t3,t4,t5,t6,ft0,ft1,ft2,ft3,ft4,ft5,ft6,ft7,fs0,fs1,fa0,fa1,fa2,fa3,fa4,fa5,fa6,fa7"
+ ",fs2,fs3,fs4,fs5,fs6,fs7,fs8,fs9,fs10,fs11,ft8,ft9,ft10,ft11,fcsr").split(",");
```

Reply:
Agreed and updated the code against the suggestion above.